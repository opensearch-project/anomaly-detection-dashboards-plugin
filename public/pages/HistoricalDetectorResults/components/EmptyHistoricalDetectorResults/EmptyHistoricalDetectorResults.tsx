/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import {
  EuiEmptyPrompt,
  EuiLink,
  EuiIcon,
  EuiButton,
  EuiOverlayMask,
} from '@elastic/eui';
import React, { Fragment, useState } from 'react';
import { Detector } from '../../../../models/interfaces';
import { BASE_DOCS_LINK } from '../../../../utils/constants';
import { HistoricalRangeModal } from '../HistoricalRangeModal';

interface EmptyHistoricalDetectorResultsProps {
  detector: Detector;
  onConfirm(startTime: number, endTime: number): void;
}

export const EmptyHistoricalDetectorResults = (
  props: EmptyHistoricalDetectorResultsProps
) => {
  const [historicalRangeModalOpen, setHistoricalRangeModalOpen] = useState<
    boolean
  >(false);
  return (
    <EuiEmptyPrompt
      title={<h2>You haven't yet set up historical analysis</h2>}
      body={
        <Fragment>
          {historicalRangeModalOpen ? (
            <EuiOverlayMask>
              <HistoricalRangeModal
                detector={props.detector}
                onClose={() => setHistoricalRangeModalOpen(false)}
                onConfirm={props.onConfirm}
                isEdit={false}
              />
            </EuiOverlayMask>
          ) : null}
          <p>
            Historical analysis lets you apply anomaly detection models over
            long historical data windows (weeks or months). You can identify
            anomaly patterns, seasonality, and trends.{' '}
            <EuiLink
              href={`${BASE_DOCS_LINK}/ad/index/#step-6-analyze-historical-data`}
              target="_blank"
            >
              Learn more &nbsp;
              <EuiIcon size="s" type="popout" />
            </EuiLink>{' '}
          </p>
        </Fragment>
      }
      actions={
        <EuiButton
          fill={true}
          onClick={() => {
            setHistoricalRangeModalOpen(true);
          }}
        >
          Run historical analysis
        </EuiButton>
      }
    />
  );
};
