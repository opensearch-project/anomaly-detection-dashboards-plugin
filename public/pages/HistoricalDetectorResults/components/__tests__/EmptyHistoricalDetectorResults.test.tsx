/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { render } from '@testing-library/react';
import { EmptyHistoricalDetectorResults } from '../EmptyHistoricalDetectorResults/EmptyHistoricalDetectorResults';

describe('<EmptyHistoricalDetectorResults /> spec', () => {
  test('renders component', () => {
    const { container, getByText } = render(
      <EmptyHistoricalDetectorResults detector={{}} onConfirm={jest.fn()} />
    );
    expect(container.firstChild).toMatchSnapshot();
    getByText(`You haven't yet set up historical analysis`);
    getByText('Run historical analysis');
  });
});
