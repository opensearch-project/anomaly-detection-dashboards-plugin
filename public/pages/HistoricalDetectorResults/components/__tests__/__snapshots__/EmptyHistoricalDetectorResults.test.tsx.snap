// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyHistoricalDetectorResults /> spec renders component 1`] = `
<div
  class="euiEmptyPrompt"
  data-test-subj="emptyHistoricalAnalysisMessage"
>
  <h2
    class="euiTitle euiTitle--medium"
  >
    You haven't yet set up historical analysis
  </h2>
  <span
    class="euiTextColor euiTextColor--subdued"
  >
    <div
      class="euiSpacer euiSpacer--m"
    />
    <div
      class="euiText euiText--medium"
    >
      <p>
        Historical analysis lets you apply anomaly detection models over long historical data windows (weeks or months). You can identify anomaly patterns, seasonality, and trends.
         
        <a
          class="euiLink euiLink--primary"
          href="https://opensearch.org/docs/monitoring-plugins/ad/index/#step-6-analyze-historical-data"
          rel="noopener noreferrer"
          target="_blank"
        >
          Learn more
          <svg
            aria-hidden="true"
            aria-label="External link"
            class="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
            focusable="false"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
            />
          </svg>
          <span
            class="euiScreenReaderOnly"
          >
            (opens in a new tab or window)
          </span>
        </a>
         
      </p>
    </div>
  </span>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <button
    class="euiButton euiButton--primary euiButton--fill"
    data-test-subj="runHistoricalAnalysisButton"
    type="button"
  >
    <span
      class="euiButtonContent euiButton__content"
    >
      <span
        class="euiButton__text"
      >
        Run historical analysis
      </span>
    </span>
  </button>
</div>
`;
